platform/migration/platform-migratable-api/src/main/java/org/codice/ddf/migration/ExportMigrationEntry.java

```diff
@@ -0,0 +1,109 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.migration;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.Optional;
+import java.util.function.BiPredicate;
+
+import org.codice.ddf.util.function.EBiConsumer;
+
+/**
+ * The <code>ExportMigrationEntry</code> interfaces provides support for artifacts that are being
```

Comment:
❓ The relationship between `MigrationContext` and `MigrationEntry` is a little confusing, especially since they both share similar methods. It might be worth providing some usage example(s) and how they can/should be used.